### PR TITLE
Fixed overflow in m_bgData.mountSpell

### PR DESCRIPTION
**Pull request by robinsch**
**Changes proposed:**
Fixed possible overflow in m_bgData.mountSpell 

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

Closes #  (insert issue tracker number)

**Tests performed:**

(Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -10853,8 +10853,30 @@ bool Unit::InitTamedPet(Pet* pet, uint8 level, uint32 spell_id)
             creature->SetLootRecipient(nullptr);
     }
 
-    if (isRewardAllowed && creature && creature->GetLootRecipient())
-        player = creature->GetLootRecipient();
+    if (isRewardAllowed && creature)
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp:10856]: are these changes supposed to be in this pr ?

File: sql/updates/characters/3.3.5/2020_09_02_00_characters.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE `character_battleground_data` 
+MODIFY COLUMN `mountSpell` int(10) UNSIGNED NOT NULL DEFAULT 0 AFTER `taxiEnd`;
```
Comment (jackpoz) [inline sql/updates/characters/3.3.5/2020_09_02_00_characters.sql:2]: From SQL checks of Circle CI:
>Checking database 'characters' for missing filenames in tables...
"sql/updates/characters/2020_09_02_00_characters.sql" is missing in the 'characters'.'updates' table.

These changes need to be applied to https://github.com/TrinityCore/TrinityCore/blob/3.3.5/sql/base/characters_database.sql and the update needs to be added there too

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (robinsch) [reply src/server/game/Entities/Unit/Unit.cpp:10856]: git history seems broken (this change is actually merged into TC). will have to reset

File: src/server/game/Entities/Player/Player.cpp [hunk 3]
```diff
@@ -25845,9 +25845,9 @@ void Player::_SaveBGData(CharacterDatabaseTransaction& trans)
     stmt->setFloat (5, m_bgData.joinPos.GetPositionZ());
     stmt->setFloat (6, m_bgData.joinPos.GetOrientation());
     stmt->setUInt16(7, m_bgData.joinPos.GetMapId());
-    stmt->setUInt16(8, m_bgData.taxiPath[0]);
-    stmt->setUInt16(9, m_bgData.taxiPath[1]);
-    stmt->setUInt16(10, m_bgData.mountSpell);
+    stmt->setUInt32(8, m_bgData.taxiPath[0]);
+    stmt->setUInt32(9, m_bgData.taxiPath[1]);
+    stmt->setUInt32(10, m_bgData.mountSpell);
```
Comment (Treeston) [inline src/server/game/Entities/Player/Player.cpp:25850]: where is this data read by the core?

Comment (robinsch) [reply src/server/game/Entities/Player/Player.cpp:25850]: void Player::_LoadBGData(PreparedQueryResult result)

Comment (Treeston) [reply src/server/game/Entities/Player/Player.cpp:25850]: what getter is used there

**Comment by robinsch** [comment]
I cannot made circleci happy and supply SQL updates because I do not have a running version of TC

**Comment by Treeston** [comment]
You cannot make circleci happy because your `characters` base file update is not sufficient. Add the row for the update file you added to `updates`' data in the base file.

**Comment by HelloKitty** [comment]
Good find!

**Comment by Shauren** [comment]
Merged, thanks

**Pull request closed without merging**